cuda_core/cuda/core/experimental/_device.py

```diff
@@ -957,26 +957,34 @@ def __new__(cls, device_id=None):
 
         # important: creating a Device instance does not initialize the GPU!
         if device_id is None:
-            device_id = handle_return(runtime.cudaGetDevice())
-            assert_type(device_id, int)
+            err, dev = driver.cuCtxGetDevice()
+            if err == 0:
+                device_id = int(dev)
+            else:
+                ctx = handle_return(driver.cuCtxGetCurrent())
```

Comment:
Is there a specific error code or set of error codes we should be handling here? If the above `driver.cuCtxGetDevice()` call returns an error that we don't expect we should probably raise it as an exception instead of it propagating to the `driver.cuCtxGetCurrent()` call?

Reply:
Invalid context (201) is now checked in commit https://github.com/NVIDIA/cuda-python/commit/6b245ff1784004816c32429ccaf6517218e03ace